R/build_lookup.R

```diff
@@ -62,6 +62,39 @@ parse_imgt_fasta <- function(infile) {
   imgt_list
 }
 
+#' Extract gene names from a reference CSV for a given species
+#'
+#' `parse_mixcr_csv()` generates mixcr gene name data for a given species.
+#' @param species A string specifying the species, one of "human", "mouse",
+#' or "rhesus".
+#'
+#' @return A character vector of gene names.
+#' @export
+#' @keywords internal
+#' @examples
+#' parse_mixcr_csv("human")
+parse_mixcr_csv <- function(species) {
+  ## Throw error if input is not one of "human", "mouse", or "rhesus"
+  if (!species %in% c("human", "mouse", "rhesus")) {
+    stop("Input must be one of \'human\', \'mouse\', or \'rhesus\'")
+  }
+
+  ## Path to folder for reference data for given species
+  directory <- paste0("inst/extdata/mixcr_ref_data/", species)
```

Comment:
We could have a check where we make sure `directory` is a valid path

Reply:
See https://github.com/seshadrilab/tcrconvertr/blob/fd9e4455a56757d3d366e71f8d6bcc3c191d1d5d/R/build_lookup.R#L169 as one way to do it